addons/project/migrations/11.0.1.1/pre-migration.py

```diff
@@ -27,9 +27,29 @@ def move_fields_from_module(env):
     )
 
 
+def avoid_subproject_constraint(env):
+    """On v11, there's a new constraint that prevents that a subtask belongs to
+    a project different from the one defined in parent project as subproject,
+    so for avoiding this constraint, we empty the subproject in the parent
+    project in these cases.
+    """
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE project_project pp
+        SET subtask_project_id = NULL
```

Comment:
Why not `SET subtask_project_id = pt.project_id`? 🤔

Reply:
Because you can several subtasks, each of them pointing to a different project, so there's no possibility of satisfying all of them.

Reply:
But why not pointing all of them to the same?

Reply:
Because then you are altering the legit project of some old tasks, which is not a fortunate technique. We only want to avoid the constraint, not to alter data. And this field is only a default value for new tasks, not something defining.

Reply:
ok